adal/src/main/java/com/microsoft/aad/adal/DefaultTokenCacheStore.java

```diff
@@ -336,11 +336,14 @@ public void clearTokensForUser(String userId) {
 
         while (results.hasNext()) {
             final TokenCacheItem tokenCacheItem = results.next();
-                if (isAboutToExpire(tokenCacheItem.getExpiresOn())) {
+            if (tokenCacheItem.getExtendedExpiresOn() != null) {
+                if (isAboutToExpire(tokenCacheItem.getExtendedExpiresOn())) {
                     tokenItems.add(tokenCacheItem);
                 }
+            } else if (isAboutToExpire(tokenCacheItem.getExpiresOn())) {
+                tokenItems.add(tokenCacheItem);
```

Comment:
hm, do we ever delete expired cached tokens? I don't think so, at least in android, we don't do so.